src/Handler/NewScream.hs

```diff
@@ -27,13 +29,22 @@ data ScreamFields = ScreamFields
     , imageField :: Maybe FileInfo
     }
 
-fromScreamFields :: ScreamFields -> Handler Scream
-fromScreamFields f = do
+parseImages :: ScreamFields -> ScreamId -> Handler [Image]
+parseImages f sid = do
+    images <- S3.uploadImages $ maybeToList $ imageField f
+    return $ fmap createImage images
```

Comment:
I dig this, but this whole thing is going to change when I add the ability to upload multiple images via the UI, so I don't know that it's worth tweaking. I'll definitely revisit this suggestion when I'm a bit further with this stuff.